File: CreaObjetos/CreaObjetos/clases/Propiedad.vb
Public Class Propiedad
    Private _nombre As String
    Private _tipo As String
    Private _isNull As Boolean
    Public Property Nombre() As String
        Get
            Return Me._nombre
        End Get
        Set(ByVal value As String)
            Me._nombre = value
        End Set
    End Property

    Public Property Tipo() As String
        Get
            Return Me._tipo
        End Get
        Set(ByVal value As String)
            Me._tipo = value
        End Set
    End Property
    Public Property isNull() As Boolean
        Get
            Return Me._isNull
        End Get
        Set(ByVal value As Boolean)
            Me._isNull = value
        End Set
    End Property
    Public ReadOnly Property titulo() As String
        Get
            Return Me.Nombre.Substring(0, 1).ToUpper() & Me.Nombre.Substring(1, Me.Nombre.Length - 1)
        End Get
    End Property
    Public ReadOnly Property abreValida() As String
        Get
            Dim abreValidas() As String = {"", "ValNulosInt(", "ValNulosInt(", "ValidaFecha(Ctype(", "Ctype(", "ValNulosInt(", "", "Byte()"}
            Dim ind As Integer = Me.dameTipo()
            If ind >= 0 Then
                Return abreValidas(ind)
            End If
            Return ""
        End Get
    End Property

    Public ReadOnly Property cierraValida() As String
        Get
            Dim tiposVB() As String = {"string", "Integer", "Decimal", "datetime", "boolean", "double", "string", "Byte()"}
            Dim cierraValidas() As String = {".Text.Length > 0", ") > 0 ", "ValNulosInt(", "ValidaFecha(Ctype(", "Ctype(", "ValNulosInt(", "", "Byte()"}
            Dim ind As Integer = Me.dameTipo()
            If ind >= 0 Then
                Return cierraValidas(ind)
            End If
            Return ""
        End Get
    End Property
    Public ReadOnly Property tipoControl() As String
        Get
            Dim tiposCtl() As String = {"txt", "txt", "txt", "dtp", "chb", "txt", "txt", "fbw", "pcb", "txt"}
            Dim ind As Integer = Me.dameTipo()
            If ind >= 0 Then
                Return tiposCtl(ind)
            End If
            Return ""

        End Get
    End Property
    Public ReadOnly Property nombreControl() As String
        Get
            Dim tiposCtl() As String = {"TextBox", "TextBox", "TextBox", "DateTimePicker", _
            "CheckBox", "TextBox", "TextBox", "fbw", "PictureBox", "TextBox"}
            Dim ind As Integer = Me.dameTipo()
            If ind >= 0 Then
                Return tiposCtl(ind)
            End If
            Return ""

        End Get
    End Property
    Public Function dameTipo() As Integer
        Dim tiposSql As New List(Of String)
        tiposSql.add("varchar")
        tiposSql.Add("int")
        tiposSql.Add("money")
        tiposSql.Add("datetime")
        tiposSql.Add("bit")
        tiposSql.Add("float")
        tiposSql.Add("nvarchar")
        tiposSql.Add("varbinary")
        tiposSql.Add("image")
        tiposSql.Add("text")
        Return tiposSql.IndexOf(Me.Tipo)
        Return -1
    End Function
    Public Function convertipoVB(ByVal tipo As String) As String
        Dim indTipo As Integer = Me.dameTipo()
        Dim tiposVB() As String = {"string", "Integer", "Decimal", "datetime", "boolean", "double", "string", "Byte()", "Byte()", "String"}

        If indTipo >= 0 Then
            Return tiposVB(indTipo)
        End If
        Return ""
    End Function
    Public Function tipoVB() As String
        
        Return convertipoVB(Me.Tipo)
    End Function

End Class

File: trunk/CreaObjetos/CreaObjetos.ConvertedToC#/clases/Propiedad.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
namespace WindowsApplication1
{
	public class Propiedad
	{
		private string _nombre;
		private string _tipo;
		private bool _isNull;
		public string Nombre {
			get { return this._nombre; }
			set { this._nombre = value; }
		}

		public string Tipo {
			get { return this._tipo; }
			set { this._tipo = value; }
		}
		public bool isNull {
			get { return this._isNull; }
			set { this._isNull = value; }
		}
		public string titulo {
			get { return this.Nombre.Substring(0, 1).ToUpper() + this.Nombre.Substring(1, this.Nombre.Length - 1); }
		}
		public string abreValida {
			get {
				string[] abreValidas = {
					"",
					"ValNulosInt(",
					"ValNulosInt(",
					"ValidaFecha(Ctype(",
					"Ctype(",
					"ValNulosInt(",
					"",
					"Byte()"
				};
				int ind = this.dameTipo();
				if (ind >= 0) {
					return abreValidas[ind];
				}
				return "";
			}
		}

		public string cierraValida {
			get {
				string[] tiposVB = {
					"string",
					"Integer",
					"Decimal",
					"datetime",
					"boolean",
					"double",
					"string",
					"Byte()"
				};
				string[] cierraValidas = {
					".Text.Length > 0",
					") > 0 ",
					"ValNulosInt(",
					"ValidaFecha(Ctype(",
					"Ctype(",
					"ValNulosInt(",
					"",
					"Byte()"
				};
				int ind = this.dameTipo();
				if (ind >= 0) {
					return cierraValidas[ind];
				}
				return "";
			}
		}
		public string tipoControl {
			get {
				string[] tiposCtl = {
					"txt",
					"txt",
					"txt",
					"dtp",
					"chb",
					"txt",
					"txt",
					"fbw",
					"pcb",
					"txt"
				};
				int ind = this.dameTipo();
				if (ind >= 0) {
					return tiposCtl[ind];
				}
				return "";

			}
		}
		public string nombreControl {
			get {
				string[] tiposCtl = {
					"TextBox",
					"TextBox",
					"TextBox",
					"DateTimePicker",
					"CheckBox",
					"TextBox",
					"TextBox",
					"fbw",
					"PictureBox",
					"TextBox"
				};
				int ind = this.dameTipo();
				if (ind >= 0) {
					return tiposCtl[ind];
				}
				return "";

			}
		}
		public int dameTipo()
		{
			List<string> tiposSql = new List<string>();
			tiposSql.Add("varchar");
			tiposSql.Add("int");
			tiposSql.Add("money");
			tiposSql.Add("datetime");
			tiposSql.Add("bit");
			tiposSql.Add("float");
			tiposSql.Add("nvarchar");
			tiposSql.Add("varbinary");
			tiposSql.Add("image");
			tiposSql.Add("text");
			return tiposSql.IndexOf(this.Tipo);
			return -1;
		}
		public string convertipoVB(string tipo)
		{
			int indTipo = this.dameTipo();
			string[] tiposVB = {
				"string",
				"Integer",
				"Decimal",
				"datetime",
				"boolean",
				"double",
				"string",
				"Byte()",
				"Byte()",
				"String"
			};

			if (indTipo >= 0) {
				return tiposVB[indTipo];
			}
			return "";
		}
		public string tipoVB()
		{

			return convertipoVB(this.Tipo);
		}

	}
}
